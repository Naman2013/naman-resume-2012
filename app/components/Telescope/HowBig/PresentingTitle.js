import React from 'react';
import PropTypes from 'prop-types';
import { monoFont } from 'styles/variables/fonts';

const PresentingTitle = ({ dimension }) => (
  <g>
    <text
      x={(dimension / 2)}
      y={(dimension - 100)}
      alignmentBaseline="middle"
      textAnchor="middle"
      className="text"
    >
      HOW BIG?
    </text>

    <style jsx>
      {`
        .text {
          font-family: ${monoFont};
          fill: aqua;
          font-size: 20px;
        }
      `}
    </style>
  </g>
);

PresentingTitle.propTypes = {
  dimension: PropTypes.number.isRequired,
};

export default PresentingTitle;
