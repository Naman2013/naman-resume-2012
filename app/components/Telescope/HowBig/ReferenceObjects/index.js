import React from 'react';

import Earth from './Earth';
import MilkyWay from './MilkyWay';
import SolarSystem from './SolarSystem';
import Sun from './Sun';

export default {
  SOLAR_SYSTEM: {
    render: props => (<Earth {...props} />),
  },
  STAR: {
    render: props => (<Sun {...props} />),
  },
  MILKY_WAY: {
    render: props => (<SolarSystem {...props} />),
  },
  DEEP_SPACE: {
    render: props => (<MilkyWay {...props} />),
  },
};
