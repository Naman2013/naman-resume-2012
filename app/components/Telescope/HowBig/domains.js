import React from 'react';
import { Enum } from 'enumify';
import ObjectFrame from './ReferenceObjects/ObjectFrame';
import nebula from './ReferenceObjects/svg/eskino-nebula.svg';

/**
  target inputs
  domain
  size of the reference

  scale - 0 to 1
  next domain is always 1
  domain comparison is some percentage of 1
*/

class Domains extends Enum {}
Domains.initEnum({
  SOLAR_SYSTEM: {
    render: props => (
      <ObjectFrame
        svgURL={nebula}
        {...props}
      />
    ),
  },
  STAR: {
    render: props => (
      <ObjectFrame
        svgURL={nebula}
        {...props}
      />
    ),
  },
  MILKY_WAY: {
    render: props => (
      <ObjectFrame
        svgURL={nebula}
        {...props}
      />
    ),
  },
  DEEP_SPACE: {
    render: props => (
      <ObjectFrame
        svgURL={nebula}
        {...props}
      />
    ),
  },
});

export default Domains;
