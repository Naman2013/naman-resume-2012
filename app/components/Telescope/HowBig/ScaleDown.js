import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ObjectFrame from './ReferenceObjects/ObjectFrame';
import FadeSVG from 'components/common/Fade/FadeSVG';
import domains from './domains';

class ScaleDown extends Component {
  static propTypes = {
    referenceObject: PropTypes.oneOf([
      'SOLAR_SYSTEM',
      'STAR',
      'MILKY_WAY',
      'DEEP_SPACE',
    ]).isRequired,
    targetObjectURL: PropTypes.string.isRequired,
    targetObjectScale: PropTypes.number.isRequired,
    onComplete: PropTypes.func.isRequired,
  };

  state = {
    targetCurrentScale: 1,
  };

  render() {
    const {
      referenceObject,
      targetObjectURL,
      dimension,
    } = this.props;

    const { targetCurrentScale } = this.state;

    return (
      <g>
        {domains.enumValueOf(referenceObject).render()}
        <ObjectFrame />
      </g>
    );
  }
}

export default ScaleDown;
