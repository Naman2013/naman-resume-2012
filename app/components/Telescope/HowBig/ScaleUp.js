import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';
import convertToDecimal from 'utils/convertToDecimal';
import calculatePercentage from 'utils/calculatePercentage';
import FadeSVG from '../../common/Fade/FadeSVG';
import SVGText from '../common/SVGText';
import ObjectFrame from './ReferenceObjects/ObjectFrame';
import domains from './domains';
import easingFunctions, { animateValues } from '../../../utils/easingFunctions';

class ScaleUp extends PureComponent {
  static PAUSE_BEFORE_MOVING_REFERENCE = 1000;

  static DURATION_TO_MOVE_REFERENCE = 500;

  static PAUSE_BEFORE_INTRODUCING_TARGET_OBJECT = 1500;

  static ANIMATE_TARGET_DURATION = 500;

  static PAUSE_BEFORE_COMPLETE = 2000;

  static ARTWORK_VS_CANVAS_SIZE_PERCENTAGE = 0.8;

  static propTypes = {
    dimension: PropTypes.number,
    targetImageURL: PropTypes.string.isRequired,
    targetObjectName: PropTypes.string.isRequired,
    domain: PropTypes.string.isRequired,
    referenceObjectScale: PropTypes.number.isRequired,
    onComplete: PropTypes.func,
  };

  static defaultProps = {
    dimension: 500,
    onComplete: noop,
  };

  state = {
    referenceObjectLoaded: false,
    targetObjectLoaded: false,
    showReference: false,
    referenceScale: 80,
    showReferenceText: true,
    referencePositionModifier: 100,
    targetObjectOpacity: 0,
    bottomTitle: '',
  };

  componentDidMount() {
    this.beginDelayToShowReference();
  }

  componentWillUnmount() {
    clearTimeout(this.timerDelayPresentReference);
    clearTimeout(this.timerDelayScaleReference);
    clearTimeout(this.timerDelayToAnimateReference);
    clearTimeout(this.timerDelayToPresentTarget);
    clearTimeout(this.timerPauseBeforeComplete);
  }

  handleReferenceObjectLoaded = () => {
    this.setState({ referenceObjectLoaded: true });
  };

  handleTargetObjectLoaded = () => {
    this.setState({ targetObjectLoaded: true });
  };

  beginDelayToShowReference() {
    setTimeout(() => {
      this.presentReference();
    }, this.props.startDelaySec * 1000);
  }

  presentReference() {
    const { changeTitle, phase1TopTitle, phase1BottomTitle } = this.props;

    this.setState({ showReference: true, bottomTitle: phase1BottomTitle });
    changeTitle(phase1TopTitle);
    this.timerDelayScaleReference = setTimeout(() => {
      this.scaleReference();
    }, this.props.phase1Sec * 1000);
  }

  scaleReference() {
    const { changeTitle, phase2TopTitle, phase2BottomTitle } = this.props;
    changeTitle(phase2TopTitle);
    this.animateScaleOfReference = animateValues(
      {
        referenceScale: this.state.referenceScale,
      },
      this.props.phase2Sec * 1000,
      {
        referenceScale: this.props.referenceObjectScale * 100,
        onUpdate: ({ referenceScale }) => {
          this.setState(() => ({
            referenceScale,
            bottomTitle: phase2BottomTitle,
          }));
        },
        onComplete: () => {
          this.prepareToAnimateReferenceLocation();
        },
        ease: easingFunctions.easeInOutQuad,
      }
    );
  }

  prepareToAnimateReferenceLocation() {
    this.timerDelayToAnimateReference = setTimeout(() => {
      this.animateMoveReference();
    }, ScaleUp.PAUSE_BEFORE_MOVING_REFERENCE);
  }

  animateMoveReference() {
    const { changeTitle, phase3TopTitle, phase3BottomTitle } = this.props;
    changeTitle(phase3TopTitle);
    this.animateReferenceMoveHandle = animateValues(
      {
        referencePositionModifier: this.state.referencePositionModifier,
      },
      this.props.phase3Sec * 1000,
      {
        referencePositionModifier: 17,
        onUpdate: ({ referencePositionModifier }) => {
          this.setState(() => ({
            referencePositionModifier,
            bottomTitle: phase3BottomTitle,
          }));
        },
        onComplete: () => {
          this.prepareToIntroduceTargetObject();
        },
        ease: easingFunctions.easeInOutQuad,
      }
    );
  }

  prepareToIntroduceTargetObject() {
    this.timerDelayPresentReference = setTimeout(() => {
      this.setState(() => ({ showReferenceText: false }));
      this.animateIntroduceTargetObject();
    }, ScaleUp.PAUSE_BEFORE_INTRODUCING_TARGET_OBJECT);
  }

  animateIntroduceTargetObject() {
    const { changeTitle, phase4TopTitle, phase4BottomTitle } = this.props;
    changeTitle(phase4TopTitle);
    this.animateTargetObjectOpacity = animateValues(
      {
        targetObjectOpacity: this.state.targetObjectOpacity,
      },
      this.props.phase4Sec * 1000,
      {
        targetObjectOpacity: 1,
        onUpdate: ({ targetObjectOpacity }) => {
          this.setState(() => ({
            targetObjectOpacity,
            bottomTitle: phase4BottomTitle,
          }));
        },
        onComplete: () => {
          this.complete();
        },
        ease: easingFunctions.easeInOutQuad,
      }
    );
  }

  complete() {
    this.timerPauseBeforeComplete = setTimeout(() => {
      this.props.onComplete();
    }, ScaleUp.PAUSE_BEFORE_COMPLETE);
  }

  timerDelayPresentReference = undefined;

  timerDelayScaleReference = undefined;

  timerDelayToAnimateReference = undefined;

  timerDelayToPresentTarget = undefined;

  timerPauseBeforeComplete = undefined;

  animateScaleOfReference = undefined;

  animateReferenceMoveHandle = undefined;

  render() {
    const { domain, targetImageURL, targetObjectName, dimension } = this.props;

    const {
      referenceObjectLoaded,
      showReference,
      referenceScale,
      referencePositionModifier,
      showReferenceText,
      bottomTitle,
    } = this.state;

    const displayReferenceObject = showReference && referenceObjectLoaded;
    const artworkDimension = dimension * 0.8;
    const midPoint = dimension / 2;
    const staticArtworkPosition = midPoint - artworkDimension / 2;
    const textLabelFontSize = dimension * 0.03;
    const showTargetObject = showReferenceText;

    const referenceSize = calculatePercentage(dimension, referenceScale);

    const referencePosition =
      (midPoint - referenceSize / 2) * (referencePositionModifier / 100);

    const domainValues = domains.enumValueOf(domain);

    return (
      <g>
        <FadeSVG isHidden={!displayReferenceObject}>
          <g
            style={{
              transformOrigin: 'center',
            }}
          >
            {domainValues.render({
              width: referenceSize,
              height: referenceSize,
              x: referencePosition,
              y: referencePosition,
              onLoadCallback: this.handleReferenceObjectLoaded,
            })}
          </g>
          <FadeSVG isHidden={!showReferenceText}>
            <SVGText
              text={bottomTitle}
              x={midPoint}
              y={dimension - dimension * 0.05}
              displayProperties={{
                fontSize: `${textLabelFontSize}px`,
              }}
            />
          </FadeSVG>
        </FadeSVG>

        <FadeSVG isHidden={showTargetObject}>
          <g>
            <ObjectFrame
              svgURL={targetImageURL}
              width={artworkDimension}
              height={artworkDimension}
              x={staticArtworkPosition}
              y={staticArtworkPosition}
              onLoadCallback={this.handleTargetObjectLoaded}
            />

            <SVGText
              text={bottomTitle}
              x={midPoint}
              y={dimension - dimension * 0.05}
              displayProperties={{ fontSize: `${textLabelFontSize}px` }}
            />
          </g>
        </FadeSVG>
      </g>
    );
  }
}

export default ScaleUp;
