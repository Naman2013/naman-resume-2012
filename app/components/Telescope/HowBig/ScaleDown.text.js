import React from 'react';
import { shallow } from 'enzyme';
import ScaleDown from './ScaleDown';

describe('ScaleUp', () => {
  it('should render correctly', () => {
    const scaleDown = shallow(<ScaleDown />);
    expect(scaleDown).toMatchSnapshot();
  });
});
