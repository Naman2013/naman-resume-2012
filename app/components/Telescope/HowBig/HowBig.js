import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SVGText from '../common/SVGText';
import ScaleUp from './ScaleUp';
import ScaleDown from './ScaleDown';

class HowBig extends Component {
  static propTypes = {
    dimension: PropTypes.number.isRequired,
    referenceObjectScale: PropTypes.number.isRequired,
    referenceObject: PropTypes.oneOf([
      'SOLAR_SYSTEM',
      'STAR',
      'MILKY_WAY',
      'DEEP_SPACE',
    ]).isRequired,
    targetObjectScale: PropTypes.number.isRequired,
    targetObjectURL: PropTypes.string.isRequired,
    onComplete: PropTypes.func.isRequired,
  };

  state = {};

  componentWillUnmount() {
    this.props.onComplete();
  }

  render() {
    const {
      dimension,
      referenceObjectScale,
      referenceObject,
      targetObjectScale,
      targetObjectURL,
      onComplete,
    } = this.props;
    const isScaledUp = (targetObjectScale > referenceObjectScale);

    return (
      <g>
        <SVGText
          text="HOW BIG?"
          x={(dimension / 2)}
          y={(dimension - 100)}
          displayProperties={{
            fontSize: '20px',
          }}
        />

        {
          (isScaledUp)
            ? <ScaleUp />
            : <ScaleDown
              dimension={dimension}
              targetObjectURL={targetObjectURL}
              targetObjectScale={targetObjectScale}
              referenceObject={referenceObject}
              onComplete={onComplete}
            />
        }
      </g>
    );
  }
}

export default HowBig;
