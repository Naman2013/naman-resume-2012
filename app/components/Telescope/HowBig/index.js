import HowBig from './HowBig';

export default HowBig;
