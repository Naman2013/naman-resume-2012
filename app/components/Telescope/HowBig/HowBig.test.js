import React from 'react';
import { shallow } from 'enzyme';
import HowBig from './HowBig';
import ScaleDown from './ScaleDown';
import ScaleUp from './ScaleUp';

describe('HowBig', () => {
  const props = {
    dimension: 100,
    referenceObjectScale: 1,
    referenceObject: 'SOLAR_SYSTEM',
    targetObjectScale: 1,
    targetObjectURL: 'https://vega.slooh.com/icons/community/human_spirit.svg',
    onComplete: jest.fn(),
  };
  const howBig = shallow(<HowBig {...props} />);

  it('should render correctly', () => {
    expect(howBig).toMatchSnapshot();
  });

  describe('when given reference and target scale', () => {
    const targetLargerProps = { referenceObjectScale: 0.34, targetObjectScale: 1 };
    const targetSmallerProps = { referenceObjectScale: 1, targetObjectScale: 0.34 };
    it('should render `<ScaleDown />` when target is smaller than the reference', () => {
      howBig.setProps(targetSmallerProps);
      expect(howBig.find(ScaleDown).length).toEqual(1);
    });

    it('should render `<ScaleUp />` when the target is larger than the reference', () => {
      howBig.setProps(targetLargerProps);
      expect(howBig.find(ScaleUp).length).toEqual(1);
    });
  });

  describe('when completing an animation', () => {
    xit('calls the `onComplete` callback provided', () => {
      expect(props.onComplete).toBeCalled();
    });
  });
});
