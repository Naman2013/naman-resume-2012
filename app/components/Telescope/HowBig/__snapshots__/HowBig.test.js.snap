// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowBig should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HowBig
    dimension={100}
    onComplete={[MockFunction]}
    referenceObject="SOLAR_SYSTEM"
    referenceObjectScale={1}
    targetObjectScale={1}
    targetObjectURL="https://vega.slooh.com/icons/community/human_spirit.svg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SVGText
          displayProperties={
            Object {
              "fontSize": "20px",
            }
          }
          text="HOW BIG?"
          x={50}
          y={0}
        />,
        <ScaleDown
          dimension={100}
          onComplete={[MockFunction]}
          referenceObject="SOLAR_SYSTEM"
          targetObjectScale={1}
          targetObjectURL="https://vega.slooh.com/icons/community/human_spirit.svg"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "displayProperties": Object {
            "fontSize": "20px",
          },
          "text": "HOW BIG?",
          "x": 50,
          "y": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dimension": 100,
          "onComplete": [MockFunction],
          "referenceObject": "SOLAR_SYSTEM",
          "targetObjectScale": 1,
          "targetObjectURL": "https://vega.slooh.com/icons/community/human_spirit.svg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "g",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SVGText
            displayProperties={
              Object {
                "fontSize": "20px",
              }
            }
            text="HOW BIG?"
            x={50}
            y={0}
          />,
          <ScaleDown
            dimension={100}
            onComplete={[MockFunction]}
            referenceObject="SOLAR_SYSTEM"
            targetObjectScale={1}
            targetObjectURL="https://vega.slooh.com/icons/community/human_spirit.svg"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "displayProperties": Object {
              "fontSize": "20px",
            },
            "text": "HOW BIG?",
            "x": 50,
            "y": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dimension": 100,
            "onComplete": [MockFunction],
            "referenceObject": "SOLAR_SYSTEM",
            "targetObjectScale": 1,
            "targetObjectURL": "https://vega.slooh.com/icons/community/human_spirit.svg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "g",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
